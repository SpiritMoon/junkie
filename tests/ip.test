#!/bin/sh

NAME="IP"
PCAPS="pcap/gre/fragmented.pcap pcap/ip6/v6inv4.pcap"
filter() {
	# filter out capture lines since capture payload sizes depends on word size
	grep -v '^Capture' |
	sed -e 's/^\([^@]\+\)@[0-9a-fx]\+\(.*\)$/\1\2/' |
	# Some inet_ntop behave differently, some favoring ":0:" others "::"
	sed -e 's/fd11:62b:ce70:1:0:5efe:a64:/fd11:62b:ce70:1::5efe:a64:/g'
}
OUTPUT="ip.out"

. "$srcdir/dump_test"
