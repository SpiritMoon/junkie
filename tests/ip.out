-- pcap/gre/fragmented.pcap --
Ethernet: head_len=14, payload=1500, vlan_id=unset, source=00:10:f3:09:19:49, dest=ac:de:48:00:10:99, proto=IPv4
IPv4: head_len=20, payload=1480, version=4, addr=80.154.67.162->192.168.80.99, proto=GRE, ttl=117, frag=Fragment, id=0x52fe, Class=0:NonECT

Ethernet: head_len=14, payload=1500, vlan_id=unset, source=00:10:f3:09:19:49, dest=ac:de:48:00:10:99, proto=IPv4
IPv4: head_len=20, payload=1480, version=4, addr=80.154.67.162->192.168.80.99, proto=GRE, ttl=117, frag=Fragment, id=0x52fe, Class=0:NonECT

Ethernet: head_len=14, payload=1188, vlan_id=unset, source=00:10:f3:09:19:49, dest=ac:de:48:00:10:99, proto=IPv4
IPv4: head_len=20, payload=1168, version=4, addr=80.154.67.162->192.168.80.99, proto=GRE, ttl=117, frag=Reassembled, id=0x52fe, Class=0:NonECT
GRE: head_len=4, payload=4124, version=0, protocol=0x800, key=None
IPv4: head_len=20, payload=4104, version=4, addr=192.168.208.10->192.168.208.18, proto=ICMP, ttl=128, frag=NoFrag, id=0x52fe, Class=0:NonECT
ICMP: head_len=4104, payload=0, type=EchoReply, err=NONE, id=512

-- pcap/ip6/v6inv4.pcap --
Ethernet: head_len=14, payload=284, vlan_id=unset, source=00:50:56:89:40:31, dest=00:50:56:89:38:7a, proto=IPv4
IPv4: head_len=20, payload=264, version=4, addr=10.100.176.236->10.100.176.71 (hashed the other way), proto=IPv6, ttl=128, frag=NoFrag, id=0x423b, Class=0:NonECT
IPv6: head_len=40, payload=224, version=6, addr=fd11:62b:ce70:1::5efe:a64:b0ec->fd11:62b:ce70:1::5efe:a64:b047 (hashed the other way), proto=TCP, ttl=128, frag=na, id=0x0000, Class=0:NonECT
TCP: head_len=20, payload=204, ports=4014(srv)->51705, flags=AckPsh, win=255, ack=3625047501, seq=967099941 (0), urg=0, opts=none

